### ISSUE TITLE
Remove dependency on external docker images

### ISSUE THREAD

**Issue by mirekphd** [elyra-ai/elyra#1349]
[This "call home"](https://github.com/elyra-ai/examples/blob/master/pipelines/dax_noaa_weather_data/analyze_NOAA_weather_data.pipeline#L17) (`docker pull` in https://github.com/elyra-ai/examples/blob/master/pipelines/dax_noaa_weather_data/analyze_NOAA_weather_data.pipeline#L17 of the "Pandas" docker image (https://hub.docker.com/r/amancevice/pandas/) from your Docker Hub account) will be unacceptable in any corporate security-restricted environment, where only security-approved docker containers and image registries are permitted (even if not running on completely air-gapped servers). In fact one should avoid any `docker pull` operations whatsoever (as this is what was assumed to work here), as these require root-level priviledges and will be unlikely to work (unless the notebook is run as root).

I might have missed it from reading the docs - please advise if there is a truly local option - running in the same docker container (supplied and approved by the corporation) where Jupyter Lab is being executed? I mean it is a false premise that this environment must be resource-poor. Jupyter client stays in a browser on a thin client machine, correct, but python kernel is nearly always run on the server-side, on a large compute node. No need to improve on this client-server arch that already works fine for individual Notebooks. Just let the user scripts (pipeline code payload) run on the same machine (but dedicated python kernel!) as the controlling notebook. This is how `papermill` works by the way (no dependency on external docker images).

**Comment by mirekphd** [comment]
@ptitzler thank you for addressing both our concerns! 

So if the docker image running Jupyter and Elyra can be completely user-defined and generic (i.e. without any custom runtime applications) and can be scanned for vulnerabilities and perhaps even hosted in an internal image registry (like Red Hat Container Registry), then it looks like "problem solved" and just a documentation issue:) I will give it a solid test drive in our on-prem Openshift installation when the time permits (which is most likely next weekend:)

**Comment by ptitzler** [comment]
Absolutely! We do publish an official Elyra container image based on the files in https://github.com/elyra-ai/elyra/tree/master/etc/docker/elyra that you could use as a baseline. Feel free to open an issue if you do run into trouble getting an image to work or reach out on https://gitter.im/elyra-ai/community. @lresende, @akchinSTC  fyi

**Comment by ptitzler** [comment]
Opened https://github.com/elyra-ai/elyra/issues/1350 to improve the content for custom Elyra container images

**Comment by mirekphd** [comment]
> 
> 
> Opened #1350 to improve the content for custom Elyra container images

Perfect, so I will report any problems I might encounter with the Elyra container under Openshift in this new issue.

**Comment by ptitzler** [comment]
I should probably mention that Elyra can be installed as part of [Open Data Hub](https://opendatahub.io/) on Red Hat OpenShift: https://elyra.readthedocs.io/en/latest/recipes/deploying-elyra-with-opendatahub.html

**Comment by mirekphd** [comment]
> 
> I should probably mention that Elyra can be installed as part of [Open Data Hub](https://opendatahub.io/) on Red Hat OpenShift: https://elyra.readthedocs.io/en/latest/recipes/deploying-elyra-with-opendatahub.html

Yes, I've heard about it from Red Hat people, but the only problem (as with Kubeflow on its own) is our obsolete Openshift installation (still running 3.11).

**Comment by ptitzler** [comment]
I turns out (my apologies) that container images built using https://github.com/elyra-ai/elyra/blob/master/etc/docker/elyra/Dockerfile won't work on RHOS. We've still got some work to do documenting how we are building the image for Open Data Hub.

**Comment by mirekphd** [comment]
> how we are building the image for Open Data Hub.
I understand you completely, it's never been easy to port containerized apps from Docker to Openshift due to additional security considerations (Jupyter Notebook was I thing the only one working out of the box:)

**Comment by TiemenSch** [comment]
I can't seem to find the "local" runtime option. Is it at all possible to use Elyra, but disable all containerization options and just use the available (Python) kernels on the host that runs JupyterLab?

**Comment by ptitzler** [comment]
> I can't seem to find the "local" runtime option. 

To use the local runtime you must create pipelines using the generic pipeline editor:

![image](https://user-images.githubusercontent.com/13068832/184374489-e3a31317-da79-4b78-b1ae-361f433cd235.png)

When you click "run" the option is displayed:

![image](https://user-images.githubusercontent.com/13068832/184374919-32154017-804a-487d-bd0c-c9cb0c5b9aaa.png)

This option is unavailable in the Kubeflow Pipelines and Airflow pipeline editor.

> Is it at all possible to use Elyra, but disable all containerization options and just use the available (Python) kernels on the host that runs JupyterLab?

Not currently. The options for Airflow and Kubeflow Pipelines are always displayed by the UI.
